### PR TITLE
Add event hub page to contributor docs

### PR DESCRIPTION
**Pull request by remidej**
### What does it do?

Adds docs about event hub to the contributor docs website

### Why is it needed?

To let Strapiers and community members know what this API is and how to use it

### How to test it?

Run the docs site locally and open `http://localhost:3000/core/utils/event-hub`
### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request

### CODE REVIEW

**Review by markkaylor — COMMENTED** [review]
Nice work Rémi!

File: docs/docs/core/utils/event-hub.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+const listener: Listener = (payload) => {};
+strapi.eventHub.on('some.event', listener);
+
+// Use its reference to remove it
+strapi.eventHub.off('some.event', listener);
+```
+
+#### `once`
+
+Registers a listener function that will only be called the first time that an event is called. Once the event has been emitted, the listener will be removed automatically. It also returns a function that can be used to remove the listener before it was called.
```
[the remark this answers is not in the document]
Comment (remidej) [reply docs/docs/core/utils/event-hub.md:124]: When you add a "once" listener there are 2 possible cases:

- you add it
- the event is emitted
- the listener is called and self destructs

or:

- you add it
- the event is NOT emitted, so the listener is not called
- you decide to remove the listener "manually" (it doesn't self destruct) using the returned function

File: docs/docs/core/utils/event-hub.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+### Emitting events
+
+#### `emit`
+
+Dispatches a new event into the hub. It returns a promise that resolves when all the subscribers have run.
+
+```ts
+// Types
+type Emit = (name: string, payload?: Object): Promise<void>;
```
[the remark this answers is not in the document]
Comment (remidej) [reply docs/docs/core/utils/event-hub.md:38]: It's true we allow anything in there. I'll update with the type `...args: any[]` which I think should be enough

**Review by markkaylor — APPROVED** [review]
Looks good to me 👍

**Review by markkaylor — APPROVED** [review]

**Comment by oiorain** [comment]
Apologies for the late review but it looks all clear to me. Well documented with code blocks. Nice work!

**Pull request merged**